### ISSUE TITLE
 load external xodr cause segmentation fault

### ISSUE THREAD

**Issue by SdKay** [carla-simulator/carla#7505]
- CARLA version: 0.9.13
- Platform/OS: ubuntu20.04
- Problem you have experienced: **xodr load segmentation fault**
- What you expected to happen: xodr load success
- Steps to reproduce:
  - test code:
```python
import sys
import os
import carla

client = carla.Client('localhost', 2000)
client.set_timeout(10.0)

with open('town01.xodr', encoding='utf-8') as od_file:
#with open('faild.xodr', encoding='utf-8') as od_file:
    try:
        data = od_file.read()
    except OSError:
        print('file could not be readed.')
        sys.exit()

vertex_distance = 2.0  
max_road_length = 500.0 
wall_height = 1.0       
extra_width = 0.6      
world = client.generate_opendrive_world(
    data, 
    carla.OpendriveGenerationParameters(
        vertex_distance=vertex_distance,
        max_road_length=max_road_length,
        wall_height=wall_height,
        additional_width=extra_width,
        smooth_junctions=True,
        enable_mesh_visibility=True
    )
)

tm = client.get_trafficmanager(8000)   # segmentation fault here!!
```
`town01.xodr` can be generate world and call `client.get_trafficmanager` normally, but `faild.xodr` cracked when call `client.get_trafficmanager`

[xodr.zip](https://github.com/carla-simulator/carla/files/15124661/xodr.zip)

**Comment by Blyron** [comment]
Could you try this with 0.9.15 or latest dev branch?

**Comment by PatrickPromitzer** [comment]
It is hard to check an openDRIVE file without a tool, because the file can get big fast.
I normally check with https://odrviewer.io/ if the file can be loaded.

I let it run through a tool I wrote to check if the roads are right.
I believe the faild.xodr file doesn't look right. (I could be wrong)
![image](https://github.com/carla-simulator/carla/assets/152880896/8c95ec81-b3b6-47c4-bc24-c0dc0995d92b)
![image](https://github.com/carla-simulator/carla/assets/152880896/de3fadaa-0c42-4541-b68c-4f3c401a565a)

**Comment by Blyron** [comment]
I mean to run your script on latest version of CARLA

**Comment by SdKay** [comment]
> Could you try this with 0.9.15 or latest dev branch?

Not yet, because we has develop lots of features based on 0.9.13 docker....Maybe I can test it on 0.9.15

**Comment by SdKay** [comment]
> It is hard to check an openDRIVE file without a tool, because the file can get big fast. I normally check with https://odrviewer.io/ if the file can be loaded.
> 
> I let it run through a tool I wrote to check if the roads are right. I believe the faild.xodr file doesn't look right. (I could be wrong) ![image](https://private-user-images.githubusercontent.com/152880896/325922912-8c95ec81-b3b6-47c4-bc24-c0dc0995d92b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQyOTIxMjMsIm5iZiI6MTcxNDI5MTgyMywicGF0aCI6Ii8xNTI4ODA4OTYvMzI1OTIyOTEyLThjOTVlYzgxLWIzYjYtNDdjNC1iYzI0LWMwZGMwOTk1ZDkyYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyOFQwODEwMjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iN2E2ZTYyNGU1MTljMzE5MTRiYzdiZGZjZDBjY2Y2MTJmNTliNzRkMjRkZDQyZTcwOGFiMWZkZjUwZTliZDFlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.aIueLwOG1hXhgA0Bs0T_7BHm4Ic7piqmdfque6KfgWQ) ![image](https://private-user-images.githubusercontent.com/152880896/325923199-de3fadaa-0c42-4541-b68c-4f3c401a565a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQyOTIxMjMsIm5iZiI6MTcxNDI5MTgyMywicGF0aCI6Ii8xNTI4ODA4OTYvMzI1OTIzMTk5LWRlM2ZhZGFhLTBjNDItNDU0MS1iNjhjLTRmM2M0MDFhNTY1YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQyOFQwODEwMjNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYmM3ZDA2ZDcyNTdjMmMyYWE4NGM3MjA2OGEwZDc2NTFiY2E1YjQ0N2YwZGU0ZWE0N2EzNzMxMmYyYzcwNzRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.R8Ind3_CjUm5XT3TCjfCQhgErzy7R2RH1PL6xQ3xRFs)

I also think there is a problem with xodr. I have also used this website to load it, but it is always in the loading state.

**Comment by PatrickPromitzer** [comment]
With the first post you told that town01.xodr can be generated, and faild.xodr can't.
With the website not be able to load it, and seeing the output of the tool I wrote, I believe the faild.xodr file it too broken to work in Carla.
With how the roads look, I would try to  make a new *xodr file, because repairing could take a lot of time.

**Comment by Blyron** [comment]
Yeah I agree with PatrickPromitzer

**Comment by Blyron** [comment]
Have you been able to verify your xodr was correct?

**Comment by SdKay** [comment]
> Have you been able to verify your xodr was correct?

Yes, there is something wrong with it, I'm trying to find out where it is and how it comes.

**Comment by Blyron** [comment]
I am closing this issue as the error is in the xodr so feel free to open it if something comes up
